java/arcs/core/util/Scheduler.kt

```diff
@@ -285,6 +290,16 @@ class Scheduler(
         ) : Scheduler.Task.Listener("dispatcher", "non-particle", block)
     }
 
+    private val currentDispatcherThreadLocal = ThreadLocal<CoroutineDispatcher?>()
+
+    /** The [Scheduler] dispatcher that the current thread is running in, or null. */
+    val currentDispatcher get() = currentDispatcherThreadLocal.get()
+
+    /** Returns true if the current thread is executing within the given [dispatcher]. */
+    fun currentlyRunningInSchedulerDispatcher(dispatcher: CoroutineDispatcher): Boolean {
```

Comment:
From my comment above, I don't think the argument is necessary. You can use `this.asCoroutineDispatcher()` and compare it against the `currentDispatcher`.